design-documents/service-isolation/invoker.md

```diff
@@ -0,0 +1,122 @@
+### Invoker API and SPI
+Responsible for contacting remote services
+#### API
+##### Remote call response
+Result of executing the call.
+Can be used by 3rd party devs to add data to remote calls results or read it when they choose to modify
+remote calling mechanism.
+```php
+interface InvokeResponseInterface {
+    /**
+     * @return mixed Value that remote service returned.
+     */
+    public function getResult();
+    
+    /**
+     * Exception that remote service has thrown, if any.
+     *
+     * @return \Throwable|null
```

Comment:
Please double check that current framework will be able to extract/hydrate \Throwable.

Reply:
even if it doesn't we can create a middleware DTO to pass exception information between services